bigquery_etl/glam/templates/extract_probe_counts_v1.sql

```diff
@@ -37,55 +47,87 @@ GROUP BY
 ),
 -- to populate total_sample for agg_type other than 'count'
 glam_sample_counts AS (
-  SELECT fsc1.os,
+  SELECT
+    fsc1.os,
     fsc1.app_version,
     fsc1.app_build_id,
     fsc1.metric,
     fsc1.key,
     fsc1.ping_type,
     fsc1.agg_type,
-    CASE WHEN fsc1.agg_type in ('max','min','sum','avg') AND fsc2.agg_type = 'count' THEN fsc2.total_sample ELSE fsc1.total_sample END as total_sample
-  FROM `{{ dataset }}.{{ prefix }}__view_sample_counts_v1`  fsc1
-  INNER JOIN `{{ dataset }}.{{ prefix }}__view_sample_counts_v1`  fsc2
-  ON fsc1.os = fsc2.os
+    CASE
+      WHEN fsc1.agg_type IN ('max', 'min', 'sum', 'avg')
+        AND fsc2.agg_type = 'count'
+        THEN fsc2.total_sample
+      ELSE fsc1.total_sample
+    END AS total_sample
+  FROM
+    `{{ dataset }}.{{ prefix }}__view_sample_counts_v1` fsc1
+  INNER JOIN
+    `{{ dataset }}.{{ prefix }}__view_sample_counts_v1` fsc2
+    ON fsc1.os = fsc2.os
     AND fsc1.app_build_id = fsc2.app_build_id
     AND fsc1.app_version = fsc2.app_version
     AND fsc1.metric = fsc2.metric
     AND fsc1.key = fsc2.key
     AND fsc1.ping_type = fsc2.ping_type
 ),
 -- get all the rcords from view_probe_counts and the matching from view_sample_counts
-ranked_data AS (SELECT
-  cp.channel,
-  cp.version,
-  cp.os,
-  cp.ping_type,
-  cp.build_id,
-  cp.build_date,
-  cp.metric,
-  cp.metric_key,
-  cp.client_agg_type,
-  cp.metric_type,
-  total_users,
-  histogram,
-  percentiles,
-  CASE WHEN client_agg_type = '' THEN 0 ELSE total_sample END AS total_sample,
-  ROW_NUMBER() OVER (PARTITION BY cp.version, cp.os, cp.build_id,cp.ping_type, cp.metric, cp.metric_key, cp.client_agg_type,cp.metric_type,histogram, percentiles
-                    ORDER BY total_users, total_sample DESC) as rnk
-FROM
-  final_probe_extract cp
-LEFT JOIN glam_sample_counts sc
-  ON
-    sc.os = cp.os
+ranked_data AS (
+  SELECT
+    cp.channel,
+    cp.version,
+    cp.os,
+    cp.ping_type,
+    cp.build_id,
+    cp.build_date,
+    cp.metric,
+    cp.metric_key,
+    cp.client_agg_type,
+    cp.metric_type,
+    total_users,
+    histogram,
+    percentiles,
+    non_norm_histogram,
+    non_norm_percentiles,
+    CASE
+      WHEN client_agg_type = ''
+        THEN 0
+      ELSE total_sample
+    END AS total_sample,
+    ROW_NUMBER() OVER (
+      PARTITION BY
+        cp.version,
+        cp.os,
+        cp.build_id,
+        cp.ping_type,
+        cp.metric,
+        cp.metric_key,
+        cp.client_agg_type,
+        cp.metric_type,
+        histogram,
+        percentiles,
+        non_norm_histogram,
```

Comment:
Added `non_norm_histogram` and `non_norm_percentiles` here

Reply:
I don't think it's necessary to have `histogram`, `percentiles`, `non_norm_histogram` or `non_norm_percentiles` here but since the normalized columns were already added, I'll leave it and once the ETL is working end to end I'll test without them.